src/main/java/frc/robot/command_groups/Shoot.java

```diff
@@ -0,0 +1,32 @@
+package frc.robot.command_groups;
+
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.subsystems.shooter.commands.FeedToShooter;
+import frc.robot.subsystems.shooter.commands.PrepareShooter;
+
+import java.util.function.DoubleSupplier;
+
+public class Shoot extends SequentialCommandGroup {
+    private Shooter shooter;
+    private DoubleSupplier distance;
+
+    public Shoot(Shooter shooter, DoubleSupplier distance) {
+        this.shooter = shooter;
+        this.distance = distance;
+        /*
+        Once the rest of the robot is operational, another command will
+        be added to feed the balls to the shooter.
+         */
+        addCommands(
+                new PrepareShooter(shooter, () -> Shooter.getSetpointVelocity(distance.getAsDouble()))
+        );
+    }
+
+    @Override
+    public void execute() {
+        if(shooter.getVelocity() == Shooter.getSetpointVelocity(distance.getAsDouble())){
+            new FeedToShooter();
+        }
```

Comment:
That's not how you do that and it's command group so you don't need execute, you should implement the commands only in the `addCommands()` function